compiler/rustc_parse/src/parser/generics.rs

```diff
@@ -253,6 +255,10 @@ impl<'a> Parser<'a> {
             } else if self.check_type() {
                 where_clause.predicates.push(self.parse_ty_where_predicate()?);
             } else {
+                if iters == 1 {
```

Comment:
I still think we need to remove this warning, since `deny(warnings)` will break CI in places that this is currently present, and we parse this without issue. This would be useful as a warning only if we would be removing this syntax later on.

Reply:
Yea, will do that

Reply:
Done.